### ISSUE TITLE
shadowPass has a n^2 performance cost

### ISSUE THREAD

**Issue by pollend** [haikarainen/kit#16]
During the shadow pass you run through all the objects in the scene looking for a light source and then iterate through the objects and build up the light map. This has a n^2 cost to performance and will have to be addressed at some point.

**Comment by haikarainen** [comment]
Yes you are completely right, this also applies to regular rendering! Literally no optimizations has been made yet (I'm a strong believer of the implement first, optimize later idiom). The first step to optimizing would be to implement frustum culling ;)

**Comment by haikarainen** [comment]
Btw, check your email
